factcast-store/src/test/java/org/factcast/store/internal/listen/PgListenerIntegrationTest.java

```diff
@@ -167,4 +169,108 @@ public void reset() {
       }
     }
   }
+
+  @Nested
+  class FactUpdateTrigger {
+
+    @Autowired PgConnectionSupplier pgConnectionSupplier;
+    @Autowired EventBus eventBus;
+    @Autowired FactStore factStore;
+
+    @AfterEach
+    @SneakyThrows
+    void unregisterListener() {
+      registerTestAsListener(
+          pgConnectionSupplier.getUnpooledConnection("test"), "UNLISTEN " + CHANNEL_FACT_UPDATE);
+    }
+
+    @Test
+    @SneakyThrows
+    void containsTransactionId() {
+      // INIT
+      var pc = pgConnectionSupplier.getUnpooledConnection("test");
+
+      // let us also register as LISTENER
+      registerTestAsListener(pc, LISTEN_UPDATE_CHANNEL_SQL);
+
+      // also register on event bus
+      var events = new UpdateNotificationEventCollector();
+      eventBus.register(events);
+
+      UUID id1 = UUID.randomUUID();
+      UUID id2 = UUID.randomUUID();
+      UUID id3 = UUID.randomUUID();
+
+      factStore.publish(
+          List.of(
+              Fact.builder()
+                  .ns("test")
+                  .type("listenerTest1")
+                  .id(id1)
+                  .version(1)
+                  .build("{\"value\":\"1\"}"),
+              Fact.builder()
+                  .ns("test")
+                  .type("listenerTest1")
+                  .id(id2)
+                  .version(2)
+                  .build("{\"value\":\"1\"}"),
+              Fact.builder()
+                  .ns("test")
+                  .type("listenerTest2")
+                  .id(id3)
+                  .version(2)
+                  .build("{\"value\":\"1\"}")));
+
+      // RUN
+      final var statement =
+          pc.prepareStatement(
+              String.format(
+                  "UPDATE fact SET payload = '{\"value\":\"2\"}' WHERE header @> '{\"type\": \"%s\"}'",
+                  "listenerTest1"));
+      statement.executeUpdate();
+
+      // ASSERT
+      // first, check trigger
```

Comment:
we're testing something different then what the code does, here. The listener assumes there was an "id" field in your notification, while here we're testing "header.id" - will change the test and fix the resulting bug .